crates/wit-component/dl/build.sh

```diff
@@ -0,0 +1,7 @@
+# This script will rebuild libdl.so from source.
+#
+# To run this, you'll need Rust nightly 2023-07-27 or later, including the
+# `wasm32-wasi` target, plus `wasi-sdk` 21 or later, installed at $WASI_SDK_PATH
+
+RUSTFLAGS="-C relocation-model=pic" cargo +nightly build -Z build-std=panic_abort,std --release --target=wasm32-wasi
```

Comment:
Could this drop the `+nightly` flag to enable running it with custom nightlies/toolchains as needed?

Also, could you update CI to have a check that runs this script and verifies that the wasm file hasn't changed?

Reply:
Also, would it make sense to change this to perhaps build with `-Copt-level=s` and/or LTO to slim things down?

Reply:
I tried a few things (`-Copt-level=s`, `-Clinker-plugin-lto`, and `#![no_std]` with a `panic_handler` that just calls `core::arch::wasm32::unreachable!()`) but the size of the output never changed at all.

Reply:
Oops -- my mistake.  I was still pointing `clang` at target/wasm32-wasi/release/libdl.a after adding `dl` to the workspace; now it's under the target dir under the root directory.  I'll try again.

Reply:
Okay, `-Copt-level=s` actually made it larger for some reason.  And I couldn't figure out the right `clang` incantation to make LTO work (kept getting `wasm-ld: error: Invalid record (Producer: 'LLVM18.1.2-rust-1.79.0-nightly' Reader: 'LLVM 17.0.6')` errors).  But `#![no_std]` with a trivial panic handler got it down to 2703 bytes.